File: pkg/dev_compiler/test/codegen_expected/language/branches_test.js
dart_library.library('language/branches_test', null, /* Imports */[
  'dart_sdk',
  'expect'
], function load__branches_test(exports, dart_sdk, expect) {
  'use strict';
  const core = dart_sdk.core;
  const dart = dart_sdk.dart;
  const dartx = dart_sdk.dartx;
  const expect$ = expect.expect;
  const branches_test = Object.create(null);
  let VoidTodynamic = () => (VoidTodynamic = dart.constFn(dart.definiteFunctionType(dart.dynamic, [])))();
  branches_test.BranchesTest = class BranchesTest extends core.Object {
    static f() {
      expect$.Expect.equals("Never reached", 0);
      return true;
    }
    static testMain() {
      let checkPointCounter = 1;
      let checkPoint1 = 0;
      let checkPoint2 = 0;
      let checkPoint3 = 0;
      let checkPoint4 = 0;
      let checkPoint5 = 0;
      let checkPoint6 = 0;
      let i = 0;
      for (let i = 0; i < 2; i++) {
        if (i == 0) {
          checkPoint1 = checkPoint1 + checkPointCounter++;
          if (true || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f()) || dart.test(branches_test.BranchesTest.f())) {
            checkPoint2 = checkPoint2 + checkPointCounter++;
          }
        } else {
          checkPoint3 = checkPoint3 + checkPointCounter++;
          if (false) {
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
            checkPoint4 = checkPointCounter++;
          }
        }
        checkPoint5 = checkPoint5 + checkPointCounter++;
      }
      checkPoint6 = checkPoint6 + checkPointCounter++;
      expect$.Expect.equals(1, checkPoint1);
      expect$.Expect.equals(2, checkPoint2);
      expect$.Expect.equals(4, checkPoint3);
      expect$.Expect.equals(0, checkPoint4);
      expect$.Expect.equals(8, checkPoint5);
      expect$.Expect.equals(6, checkPoint6);
    }
  };
  dart.setSignature(branches_test.BranchesTest, {
    statics: () => ({
      f: dart.definiteFunctionType(core.bool, []),
      testMain: dart.definiteFunctionType(dart.void, [])
    }),
    names: ['f', 'testMain']
  });
  branches_test.main = function() {
    branches_test.BranchesTest.testMain();
  };
  dart.fn(branches_test.main, VoidTodynamic());
  // Exports:
  exports.branches_test = branches_test;
});

File: tests/language_2/unsorted/branches_test.dart
// Copyright (c) 2011, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.
// Dart test for branches. Make sure that shortcuts work, even if they have
// to jump over several expressions.

import "package:expect/expect.dart";

class BranchesTest {
  static bool f() {
    Expect.equals("Never reached", 0);
    return true;
  }

  static void testMain() {
    int checkPointCounter = 1;
    int checkPoint1 = 0;
    int checkPoint2 = 0;
    int checkPoint3 = 0;
    int checkPoint4 = 0;
    int checkPoint5 = 0;
    int checkPoint6 = 0;
    int i = 0;
    for (int i = 0; i < 2; i++) {
      if (i == 0) {
        checkPoint1 += checkPointCounter++;
        if (true || // Test branch-if-true.
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f() ||
            f()) {
          checkPoint2 += checkPointCounter++;
        }
      } else {
        // Test branch (jumping over the else branch).
        checkPoint3 += checkPointCounter++;
        if (false) {
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
          checkPoint4 = checkPointCounter++; // Never reached.
        }
      }
      checkPoint5 += checkPointCounter++;
    }
    checkPoint6 += checkPointCounter++;
    Expect.equals(1, checkPoint1);
    Expect.equals(2, checkPoint2);
    Expect.equals(4, checkPoint3);
    Expect.equals(0, checkPoint4);
    Expect.equals(8, checkPoint5);
    Expect.equals(6, checkPoint6);
  }
}

main() {
  BranchesTest.testMain();
}
